pubsub.go

```diff
@@ -185,10 +187,20 @@ func (p *PubsubValueStore) Subscribe(key string) error {
 	//
 	// Also, make sure to do this *before* subscribing.
 	myID := p.host.ID()
-	_ = p.ps.RegisterTopicValidator(topic, func(ctx context.Context, src peer.ID, msg *pubsub.Message) bool {
-		cmp := p.compare(key, msg.GetData())
+	_ = p.ps.RegisterTopicValidator(topic, func(
+		ctx context.Context,
+		src peer.ID,
+		msg *pubsub.Message,
+	) pubsub.ValidationResult {
+		cmp, valid := p.compare(key, msg.GetData())
+		if !valid {
+			return pubsub.ValidationReject
+		}
```

Comment:
This seems fine are we using the scoring function at all? What do you think about rejecting `cmp < 0` as well? This should mean peers will occasionally get published, but should not get punished repeatedly. I don't know how long it takes to recover score after being penalized.